### DISCUSSION TITLE
struggling with my first rule

### DISCUSSION THREAD

**Discussion by DanielRuf** [rectorphp/rector#8785] [Q&A] [answered]
See also https://github.com/rectorphp/rector/issues/8784

# Question

I'm trying to change some code like that:

```php
use \Namespace\ClassD;

ClassD::getA();
ClassD::getA(1,2,3);
```

to this:

```php
use \Namespace\ClassC;

ClassC::getB()->getValue();
ClassC::getB()->getValue(1, null, 2,3);
```

I tried to change `name` and `class` of `StaticCall::class` but I can not get the args. And I don't know how to change or add the `use` statement.

Any pointers for the smallest possible solution?

**Comment by samsonasik** [comment, marked as answer]
Ref original answer https://github.com/rectorphp/rector/issues/8784#issuecomment-2286080304 so no need write down answer.

First, apply rename class + method + auto import https://getrector.com/demo/a38638ff-7f36-4b24-9d49-dc7a9daaff80

per AST structure at https://getrector.com/ast/a4686a7810518c6c740f126e056bdb77c39b4847

You can use MethodCall, which verify its `var` is `StaticCall`, it read from the back to front to verify it, then you can change/insert new Arg between.
